app/commands/v2/unpublish.rb

```diff
@@ -2,22 +2,67 @@ module Commands
   module V2
     class Unpublish < BaseCommand
       def call
+        validate
+        State.supersede(previous_item) if previous_item
+        transition_state
+        delete_linkable
+
+        after_transaction_commit do
+          send_downstream
+        end
+
+        Success.new(content_id: content_id)
+      end
+
+    private
+
+      def transition_state
+        raise_invalid_type unless valid_type?
+
+        begin
+          unpublish
+        rescue ActiveRecord::RecordInvalid => e
+          raise_command_error(422, e.message, fields: {})
+        end
+      end
+
+      def valid_type?
+        %w(withdrawal redirect gone vanish).include?(type)
+      end
+
+      def type
+        payload.fetch(:type)
+      end
+
+      def raise_invalid_type
+        message = "#{type} is not a valid unpublishing type"
+        raise_command_error(422, message, fields: {})
+      end
+
+      def content_item
+        @content_item ||= find_unpublishable_content_item
```

Comment:
I think it would be better remove a layer of indirection, and this method just by moving this test and assignment inside find_unpublishable_content_item.